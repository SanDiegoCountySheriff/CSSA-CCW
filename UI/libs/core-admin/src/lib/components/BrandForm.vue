<template>
  <div>
    <v-container
      fluid
      class="form-container blue-grey lighten-3 elevation-3 rounded"
    >
      <v-card
        class="form-card rounded elevation-2"
        :class="{ 'dark-card': $vuetify.theme.dark }"
      >
        <v-form
          @submit.prevent="getFormValues"
          v-model="valid"
        >
          <BrandLogo />
          <v-row>
            <v-col
              cols="12"
              sm="6"
              md="3"
            >
              <BrandTitle />
            </v-col>
          </v-row>
          <v-row class="ml-5">
            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-btn
                :disabled="!valid"
                color="success"
                class="mr-4"
                type="submit"
              >
                {{ $t('Publish') }}
              </v-btn>
            </v-col>
          </v-row>
        </v-form>
      </v-card>
    </v-container>
  </div>
</template>

<script setup lang="ts">
import BrandLogo from './BrandLogo.vue';
import BrandTitle from './BrandTitle.vue';
import formJson from '@shared-utils/formJson';
import { ref } from 'vue';

const valid = ref(true);

function getFormValues(submitEvent) {
  // Push model JSON to API endpoint for Public App consumption"
  console.log(formJson(submitEvent.target));
}
</script>
