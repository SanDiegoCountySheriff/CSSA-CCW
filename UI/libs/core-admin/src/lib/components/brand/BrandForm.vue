<!-- eslint-disable @intlify/vue-i18n/no-raw-text -->
<template>
  <v-container class="brand-form-container">
    <v-snackbar
      :value="snackbar"
      :timeout="2000"
      app
      absolute
      bottom
      color="primary"
      left
      text
    >
      {{ $t('Updated settings') }} <strong>{{ $t('successfully.') }}</strong>
    </v-snackbar>
    <v-container
      v-if="isLoading && !isError"
      fluid
    >
      <v-skeleton-loader
        fluid
        class="fill-height"
        type="list-item,
              divider, list-item-three-line,
              card-heading, image, image, image,
              image, actions"
      >
      </v-skeleton-loader>
    </v-container>
    <v-stepper
      v-else
      v-model="stepIndex"
      class="elevation-0 pb-0"
      vertical
    >
      <v-stepper-step
        :complete="stepIndex > 1"
        editable
        step="1"
      >
        Agency
      </v-stepper-step>

      <v-stepper-content step="1">
        <AgencyFormStep
          :handle-next-step="handleNextStep"
          :handle-back-step="handleBackStep"
          :handle-reset-step="handleResetStep"
        />
      </v-stepper-content>

      <v-stepper-step
        :complete="stepIndex > 2"
        editable
        step="2"
      >
        Assets
      </v-stepper-step>

      <v-stepper-content step="2">
        <AssetsFormStep
          :handle-next-step="handleNextStep"
          :handle-back-step="handleBackStep"
          :handle-reset-step="handleResetStep"
        />
      </v-stepper-content>

      <v-stepper-step
        :complete="stepIndex > 3"
        editable
        step="3"
      >
        Color Scheme
      </v-stepper-step>

      <v-stepper-content step="3">
        <ColorSchemeFormStep
          :handle-next-step="handleNextStep"
          :handle-back-step="handleBackStep"
          :handle-reset-step="handleResetStep"
        />
      </v-stepper-content>

      <v-stepper-step
        :complete="stepIndex > 4"
        editable
        step="4"
      >
        {{ $t('Configuration') }}
      </v-stepper-step>
      <v-stepper-content step="4">
        <ConfigurationFormStep
          :handle-next-step="handleNextStep"
          :handle-back-step="handleBackStep"
          :handle-reset-step="handleResetStep"
        />
      </v-stepper-content>

      <v-stepper-step
        editable
        step="5"
      >
        {{ $t('Fees') }}
      </v-stepper-step>
      <v-stepper-content step="5">
        <FeesFormStep
          :handle-next-step="handleNextStep"
          :handle-back-step="handleBackStep"
          :handle-reset-step="handleResetStep"
        />
      </v-stepper-content>
    </v-stepper>
  </v-container>
</template>

<script setup lang="ts">
import AgencyFormStep from './steps/AgencyFormStep.vue'
import AssetsFormStep from './steps/AssetsFormStep.vue'
import ColorSchemeFormStep from './steps/ColorSchemeFormStep.vue'
import ConfigurationFormStep from './steps/ConfigurationFormStep.vue'
import FeesFormStep from './steps/FeesFormStep.vue'
import { ref } from 'vue'
import { useBrandStore } from '@shared-ui/stores/brandStore'
import { useQuery } from '@tanstack/vue-query'

const stepIndex = ref(1)
const snackbar = ref(false)

const brandStore = useBrandStore()
const { isLoading, isError } = useQuery(
  ['brandSetting'],
  brandStore.getBrandSettingApi
)

function handleNextStep() {
  snackbar.value = true
  stepIndex.value++
}

function handleBackStep() {
  stepIndex.value--
}

function handleResetStep() {
  stepIndex.value = 0
}
</script>

<style lang="scss" scoped>
.sub-header {
  font-size: 1.5rem;
}
</style>
