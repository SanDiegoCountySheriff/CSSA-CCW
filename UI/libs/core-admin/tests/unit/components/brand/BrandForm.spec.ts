import BrandForm from '@core-admin/components/brand/BrandForm.vue';
import Vuetify from 'vuetify';
import { createTestingPinia } from '@pinia/testing';
import { useQuery } from '@tanstack/vue-query';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();
const pinia = createTestingPinia();
const tMock = {
  $t: text => text,
};

jest.mock('@tanstack/vue-query', () => ({
  useQuery: jest.fn(),
}));

describe('BrandForm', () => {
  let vuetify;
  let result;

  beforeEach(() => {
    vuetify = new Vuetify();
    result = {
      data: null,
      isLoading: true,
    };
    (useQuery as jest.Mock).mockImplementation(() => result);
  });

  it('Should render logo label', () => {
    result = {
      isLoading: false,
    };
    const wrapper = mount(BrandForm, {
      localVue,
      vuetify,
      pinia,
      mocks: tMock,
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('.v-input').exists()).toBe(true);
  });
});
