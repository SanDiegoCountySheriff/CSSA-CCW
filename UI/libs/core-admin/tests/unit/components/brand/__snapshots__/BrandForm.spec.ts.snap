// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandForm Should render logo label 1`] = `
<div class="container brand-form-container">
  <div class="v-snack v-snack--absolute v-snack--bottom v-snack--left v-snack--text">
    <transition-stub name="v-snack-transition">
      <div class="v-snack__wrapper v-sheet theme--light primary--text" style="display: none;">
        <div role="status" aria-live="polite" class="v-snack__content">
          Updated settings <strong>successfully.</strong></div>
        <div class="v-snack__action "></div>
      </div>
    </transition-stub>
  </div>
  <!---->
  <div class="v-stepper elevation-0 v-stepper--vertical v-sheet theme--light" else="">
    <div class="v-subheader sub-header font-weight-bold theme--light">
      System Settings
    </div>
    <div tabindex="0" class="v-stepper__step v-stepper__step--editable v-stepper__step--inactive"><span class="v-stepper__step__step">1</span>
      <div class="v-stepper__label">
        Agency
      </div>
    </div>
    <transition-stub name="tab-transition">
      <div class="v-stepper__content">
        <div class="v-stepper__wrapper" style="height: 0px;">
          <div class="mb-8 ml-2 mr-2 mt-6 elevation-3 v-card v-sheet theme--light lighten-1" style="height: 100%;" handle-back-step="function handleBackStep() {
      stepIndex.value--;
    }">
            <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="0" class="v-progress-linear v-progress-linear--absolute v-progress-linear--visible theme--light" style="bottom: 0px; height: 0px;">
              <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 0%; width: 100%;"></div>
              <div class="v-progress-linear__buffer" style="height: 0px;"></div>
              <transition-stub name="slide-x-transition">
                <div class="v-progress-linear__determinate primary" style="width: 0%;"></div>
              </transition-stub>
            </div>
            <form novalidate="novalidate" class="v-form ml-4">
              <div class="row">
                <div class="col-md-6 col-12">
                  <div class="v-input theme--light v-text-field">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-14" class="v-label theme--light" style="left: 0px; position: absolute;">Agency Name</label><input required="required" id="input-14" type="text"></div>
                        <div class="v-input__append-inner">
                          <div></div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-md-6 col-12">
                  <div class="v-input theme--light v-text-field">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-18" class="v-label theme--light" style="left: 0px; position: absolute;">Agency Sheriff Name</label><input required="required" id="input-18" type="text"></div>
                        <div class="v-input__append-inner">
                          <div></div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-md-6 col-12">
                  <div class="v-input pl-3 theme--light v-text-field">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-22" class="v-label theme--light" style="left: 0px; position: absolute;">Chief of Police Name</label><input id="input-22" type="text"></div>
                        <div class="v-input__append-inner">
                          <div></div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row justify-space-between">
                <div justify="space-between" class="col-sm-6 col-12"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Cancel
        </span></button></div>
                <div class="col-sm-6 col-12"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Publish
        </span></button></div>
              </div>
              <div class="spacer"></div>
            </form>
          </div>
        </div>
      </div>
    </transition-stub>
    <div tabindex="0" class="v-stepper__step v-stepper__step--editable v-stepper__step--inactive"><span class="v-stepper__step__step">2</span>
      <div class="v-stepper__label">
        Assets
      </div>
    </div>
    <transition-stub name="tab-transition">
      <div class="v-stepper__content">
        <div class="v-stepper__wrapper" style="height: 0px;">
          <div class="mb-8 ml-2 mr-2 mt-6 elevation-3 v-card v-sheet theme--light lighten-1" style="height: 100%;">
            <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="0" class="v-progress-linear v-progress-linear--absolute v-progress-linear--visible theme--light" style="bottom: 0px; height: 0px;">
              <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 0%; width: 100%;"></div>
              <div class="v-progress-linear__buffer" style="height: 0px;"></div>
              <transition-stub name="slide-x-transition">
                <div class="v-progress-linear__determinate primary" style="width: 0%;"></div>
              </transition-stub>
            </div>
            <form novalidate="novalidate" class="v-form ml-4">
              <div class="row">
                <div class="col-sm-6 col-12">
                  <div class="v-input ml-5 theme--light v-text-field v-text-field--placeholder v-file-input">
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-36" class="v-label theme--light" style="left: 0px; position: absolute;">Agency Logo</label>
                          <div class="v-file-input__text v-file-input__text--placeholder"></div><input accept="image/png, image/jpeg" required="required" id="input-36" type="file">
                        </div>
                        <div class="v-input__append-inner">
                          <div></div>
                        </div>
                        <div class="v-input__append-inner">
                          <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-camera theme--light"></i></div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                        <div class="v-counter theme--light">0 files (0 B in total)</div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="col"><img alt="Agency logo"></div>
              </div>
              <div class="row">
                <div class="col-sm-6 col-12">
                  <div class="v-input ml-5 theme--light v-text-field v-text-field--placeholder v-file-input">
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-40" class="v-label theme--light" style="left: 0px; position: absolute;">Agency Landing page image</label>
                          <div class="v-file-input__text v-file-input__text--placeholder"></div><input accept="image/png, image/jpeg" required="required" id="input-40" type="file">
                        </div>
                        <div class="v-input__append-inner">
                          <div></div>
                        </div>
                        <div class="v-input__append-inner">
                          <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-camera theme--light"></i></div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                        <div class="v-counter theme--light">0 files (0 B in total)</div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="col"><img alt="Agency landing page image"></div>
              </div>
              <div class="row justify-space-between">
                <div class="col-sm-6 col-12"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Cancel
        </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Back
        </span></button></div>
                <div class="col-sm-6 col-12"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Publish
        </span></button></div>
              </div>
              <div class="spacer"></div>
            </form>
          </div>
        </div>
      </div>
    </transition-stub>
    <div tabindex="0" class="v-stepper__step v-stepper__step--editable v-stepper__step--inactive"><span class="v-stepper__step__step">3</span>
      <div class="v-stepper__label">
        Color Scheme
      </div>
    </div>
    <transition-stub name="tab-transition">
      <div class="v-stepper__content">
        <div class="v-stepper__wrapper" style="height: 0px;">
          <div class="mb-8 ml-2 mr-2 mt-6 elevation-3 v-card v-sheet theme--light lighten-1" style="height: 100%;">
            <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="0" class="v-progress-linear v-progress-linear--absolute v-progress-linear--visible theme--light" style="bottom: 0px; height: 0px;">
              <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 0%; width: 100%;"></div>
              <div class="v-progress-linear__buffer" style="height: 0px;"></div>
              <transition-stub name="slide-x-transition">
                <div class="v-progress-linear__determinate primary" style="width: 0%;"></div>
              </transition-stub>
            </div>
            <form novalidate="novalidate" class="v-form ml-4">
              <div class="row">
                <div class="col-md-6 col-12">
                  <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--placeholder">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-asterisk theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-55" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Primary Theme Color</label><input required="required" id="input-55" type="text"></div>
                        <div class="v-input__append-inner">
                          <div class="v-input__icon v-input__icon--clear"><button type="button" aria-label="clear icon" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-close theme--light"></button></div>
                        </div>
                        <div class="v-input__append-inner">
                          <div class="v-menu">
                            <div style="background-color: rgb(25, 118, 210); cursor: pointer; height: 30px; width: 30px; margin-bottom: 2px; border-radius: 4px; transition: border-radius 200ms ease-in-out;"></div>
                            <!---->
                          </div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="shrink col-md-6 col-12" style="min-width: 220px;">
                  <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--placeholder">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-asterisk theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-61" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Secondary Theme Color</label><input required="required" id="input-61" type="text"></div>
                        <div class="v-input__append-inner">
                          <div class="v-input__icon v-input__icon--clear"><button type="button" aria-label="clear icon" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-close theme--light"></button></div>
                        </div>
                        <div class="v-input__append-inner">
                          <div class="v-menu">
                            <div style="background-color: rgb(66, 66, 66); cursor: pointer; height: 30px; width: 30px; margin-bottom: 2px; border-radius: 4px; transition: border-radius 200ms ease-in-out;"></div>
                            <!---->
                          </div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row justify-space-between">
                <div class="col-sm-6 col-12"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Cancel
        </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Back
        </span></button></div>
                <div class="col-sm-6 col-12"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Publish
        </span></button></div>
              </div>
              <div class="spacer"></div>
            </form>
          </div>
        </div>
      </div>
    </transition-stub>
    <div tabindex="0" class="v-stepper__step v-stepper__step--editable v-stepper__step--inactive"><span class="v-stepper__step__step">4</span>
      <div class="v-stepper__label">
        Configuration
      </div>
    </div>
    <transition-stub name="tab-transition">
      <div class="v-stepper__content">
        <div class="v-stepper__wrapper" style="height: 0px;">
          <div class="mb-8 ml-2 mr-2 mt-6 elevation-3 v-card v-sheet theme--light lighten-1" style="height: 100%;">
            <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="0" class="v-progress-linear v-progress-linear--absolute v-progress-linear--visible theme--light" style="bottom: 0px; height: 0px;">
              <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 0%; width: 100%;"></div>
              <div class="v-progress-linear__buffer" style="height: 0px;"></div>
              <transition-stub name="slide-x-transition">
                <div class="v-progress-linear__determinate primary" style="width: 0%;"></div>
              </transition-stub>
            </div>
            <form novalidate="novalidate" class="v-form ml-4">
              <div class="row">
                <div class="col-md-6 col-12">
                  <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-78" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Refresh Token Time (in minutes)</label><input required="required" id="input-78" type="number"></div>
                        <div class="v-input__append-inner">
                          <div class="v-input__icon v-input__icon--clear"><button type="button" aria-label="clear icon" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-close theme--light"></button></div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-md-6 col-12">
                  <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-83" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Payment URL</label><input required="required" id="input-83" type="text"></div>
                        <div class="v-input__append-inner">
                          <div class="v-input__icon v-input__icon--clear"><button type="button" aria-label="clear icon" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-close theme--light"></button></div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row justify-space-between">
                <div class="col-sm-6 col-12"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Cancel
        </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Back
        </span></button></div>
                <div class="col-sm-6 col-12"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Publish
        </span></button></div>
              </div>
              <div class="spacer"></div>
            </form>
          </div>
        </div>
      </div>
    </transition-stub>
    <div tabindex="0" class="v-stepper__step v-stepper__step--editable v-stepper__step--inactive"><span class="v-stepper__step__step">5</span>
      <div class="v-stepper__label">
        Fees
      </div>
    </div>
    <transition-stub name="tab-transition">
      <div class="v-stepper__content">
        <div class="v-stepper__wrapper" style="height: 0px;">
          <div class="mb-8 ml-2 mr-2 mt-6 elevation-3 v-card v-sheet theme--light lighten-1" style="height: 100%;">
            <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="0" class="v-progress-linear v-progress-linear--absolute v-progress-linear--visible theme--light" style="bottom: 0px; height: 0px;">
              <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 0%; width: 100%;"></div>
              <div class="v-progress-linear__buffer" style="height: 0px;"></div>
              <transition-stub name="slide-x-transition">
                <div class="v-progress-linear__determinate primary" style="width: 0%;"></div>
              </transition-stub>
            </div>
            <form novalidate="novalidate" class="v-form ml-4">
              <div class="row">
                <div class="col-md-6 col-12">
                  <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--prefix v-text-field--placeholder">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-99" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Standard Cost</label>
                          <div class="v-text-field__prefix">$</div><input required="required" id="input-99" type="number">
                        </div>
                        <div class="v-input__append-inner">
                          <div class="v-input__icon v-input__icon--clear"><button type="button" aria-label="clear icon" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-close theme--light"></button></div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-md-6 col-12">
                  <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--prefix v-text-field--placeholder">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-104" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Judicial Cost</label>
                          <div class="v-text-field__prefix">$</div><input required="required" id="input-104" type="number">
                        </div>
                        <div class="v-input__append-inner">
                          <div class="v-input__icon v-input__icon--clear"><button type="button" aria-label="clear icon" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-close theme--light"></button></div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-md-6 col-12">
                  <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--prefix v-text-field--placeholder">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-109" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Reserve Cost</label>
                          <div class="v-text-field__prefix">$</div><input required="required" id="input-109" type="number">
                        </div>
                        <div class="v-input__append-inner">
                          <div class="v-input__icon v-input__icon--clear"><button type="button" aria-label="clear icon" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-close theme--light"></button></div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-md-6 col-12">
                  <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--prefix v-text-field--placeholder">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-114" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Credit Fee</label>
                          <div class="v-text-field__prefix">$</div><input required="required" id="input-114" type="number">
                        </div>
                        <div class="v-input__append-inner">
                          <div class="v-input__icon v-input__icon--clear"><button type="button" aria-label="clear icon" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-close theme--light"></button></div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-md-6 col-12">
                  <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--prefix v-text-field--placeholder">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-119" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Convenience Fee</label>
                          <div class="v-text-field__prefix">$</div><input required="required" id="input-119" type="number">
                        </div>
                        <div class="v-input__append-inner">
                          <div class="v-input__icon v-input__icon--clear"><button type="button" aria-label="clear icon" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-close theme--light"></button></div>
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row justify-space-between">
                <div class="col-sm-6 col-12"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Cancel
        </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Back
        </span></button></div>
                <div class="col-sm-6 col-12"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
          Publish
        </span></button></div>
              </div>
              <div class="spacer"></div>
            </form>
          </div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>
`;
